src/runtime/arc.ts

```diff
@@ -91,7 +93,7 @@ export class Arc {
   loadedParticleInfo = new Map<string, {spec: ParticleSpec, stores: Map<string, StorageProviderBase>}>();
   readonly pec: ParticleExecutionHost;
 
-  constructor({id, context, pecFactory, slotComposer, loader, storageKey, storageProviderFactory, speculative, innerArc, stub, inspectorFactory} : ArcOptions) {
+  constructor({id, context, pecFactory, slotComposer, loader, storageKey, storageProviderFactory, speculative, innerArc, stub, inspectorFactory, javaPort} : ArcOptions) {
```

Comment:
likewise, a list of ports rather than specific options would be better. This might require some thinking about port management, so OK to defer until after this CL.

Reply:
replaced with list of ports here too - for now it'll be up to the shell to pass the right list of ports (in fact just the java port for now)